cache/lru_cache.cc

```diff
@@ -241,17 +259,33 @@ void LRUCacheShard::LRU_Insert(LRUHandle* e) {
     e->prev->next = e;
     e->next->prev = e;
     e->SetInHighPriPool(true);
+    e->SetInLowPriPool(false);
     high_pri_pool_usage_ += e->total_charge;
     MaintainPoolSize();
-  } else {
-    // Insert "e" to the head of low-pri pool. Note that when
-    // high_pri_pool_ratio is 0, head of low-pri pool is also head of LRU list.
+  } else if (low_pri_pool_ratio_ > 0 && (e->IsLowPri() || e->HasHit())) {
```

Comment:
So I'm fairly sure this is the line which causes the regression (see #10434 ). I think we have a bug causing priority inversion here: if there is no high-priority pool configured but there is a low-priority pool, high-priority items can end up in the bottom-priority pool. If so, it should be an easy fix